ocrd_network/ocrd_network/processing_worker.py

```diff
@@ -0,0 +1,317 @@
+    def process_message(self, processing_message: OcrdProcessingMessage) -> None:
+        # Verify that the processor name in the processing message
+        # matches the processor name of the current processing worker
+        if self.processor_name != processing_message.processor_name:
+            raise ValueError(f'Processor name is not matching. Expected: {self.processor_name},'
+                             f'Got: {processing_message.processor_name}')
+
+        # This can be path if invoking `run_processor`
+        # but must be ocrd.Workspace if invoking `run_cli`.
+        workspace_path = processing_message.path_to_mets
+
+        # Build the workspace from the workspace_path
+        workspace = Resolver().workspace_from_url(workspace_path)
+
+        page_id = processing_message.page_id
+        input_file_grps = processing_message.input_file_grps
+        output_file_grps = processing_message.output_file_grps
+        parameter = processing_message.parameters
+        job_id = processing_message.job_id
+
+        self.set_job_state(job_id, StateEnum.running)
+        if self.processor_class:
+            self.log.debug(f'Invoking the pythonic processor: {self.processor_name}')
+            return_status = self.run_processor_from_worker(
+                processor_class=self.processor_class,
+                workspace=workspace,
+                page_id=page_id,
+                input_file_grps=input_file_grps,
+                output_file_grps=output_file_grps,
+                parameter=parameter
+            )
+        else:
+            self.log.debug(f'Invoking the cli: {self.processor_name}')
+            return_status = self.run_cli_from_worker(
+                executable=self.processor_name,
+                workspace=workspace,
+                page_id=page_id,
+                input_file_grps=input_file_grps,
+                output_file_grps=output_file_grps,
+                parameter=parameter
+            )
+        job_state = StateEnum.success if return_status else StateEnum.failed
+        self.set_job_state(job_id, job_state)
+
+        # If the result_queue field is set, send the job status to a result queue
+        if processing_message.result_queue:
+            if self.rmq_publisher is None:
+                self.connect_publisher()
+
+            # create_queue method is idempotent - nothing happens if
+            # a queue with the specified name already exists
+            self.rmq_publisher.create_queue(queue_name=processing_message.result_queue)
+            self.log.info(f'Publishing result message to queue: {processing_message.result_queue}')
+            result_message = OcrdResultMessage(
+                job_id=str(job_id),
+                status=job_state.value,
+                # Either path_to_mets or workspace_id must be set (mutually exclusive)
+                path_to_mets=processing_message.path_to_mets,
+                workspace_id=None
```

Comment:
Expect type `str`, got `None` instead.

Reply:
@tdoan2010, how can I reproduce this?

Reply:
Are you using PyCharm? The code is highlighted there. Otherwise, it's easy to see that `None` is passed when `workspace_id` is annotated with `str`.

Reply:
Yes, and I get no highlights there. Probably I don't have some checks enabled and that's a warning for potential unexpected behavior. Inside the `OcrdProcessingMessage` the `workspace_id` is also annotated with `str` but `None` is passed by default. Is that highlighted as well?

Reply:
No, it doesn't warn for default value. But warning come when it is used.